testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -623,27 +641,95 @@ public void testOutputColumnsForCreateTableAsSelect()
+    }
+
+    @Test
+    public void testOutputColumnsForCreateTableAsSelectWithCountAll()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table_with_aggregation(aliased_aggregation) AS SELECT count(*) FROM orders GROUP BY custkey", 2);
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+        assertThat(event.getIoMetadata().getOutput().get().getColumns().get())
+                .containsExactly(new OutputColumnMetadata("aliased_aggregation", ImmutableSet.of()));
+    }
+
+    @Test
+    public void testOutputColumnsForCreateTableAsSelectWithWindowFunction()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table_with_aggregation(running_average) AS " +
+                        "SELECT avg(totalprice) OVER (PARTITION BY custkey ORDER BY orderdate ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) FROM orders",
+                2);
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+        assertThat(event.getIoMetadata().getOutput().get().getColumns().get())
+                .containsExactly(
+                        new OutputColumnMetadata(
+                                "running_average",
+                                ImmutableSet.of(
+                                        new ColumnDetail("tpch", "tiny", "orders", "totalprice"),
+                                        new ColumnDetail("tpch", "tiny", "orders", "custkey"),
+                                        new ColumnDetail("tpch", "tiny", "orders", "orderdate"))));
+    }
+
+    @Test
+    public void testOutputColumnsForCreateTableAsSelectWithUnCorrelatedQueries()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table_with_un_correlated_queries AS SELECT custkey, (SELECT nationkey FROM nation LIMIT 1) as col2 FROM orders",
+                2);
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+        assertThat(event.getIoMetadata().getOutput().get().getColumns().get())
+                .containsExactly(
+                        new OutputColumnMetadata("custkey", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "custkey"))),
+                        new OutputColumnMetadata("col2", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "nation", "nationkey"))));
+    }
+
+    @Test
+    public void testOutputColumnsForCreateTableAsSelectWithCorrelatedQueries()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table_with_un_correlated_queries AS SELECT name, (SELECT sum(acctbal) FROM supplier WHERE supplier.nationkey=nation.nationkey) as col2 FROM nation",
```

Comment:
Please extract some utility assertion method
```
assertColumnLinage(
    "SELECT name, (SELECT sum(acctbal) FROM supplier WHERE supplier.nationkey=nation.nationkey) as col2 FROM nation",
    columnLinage("name", "tpch.tiny.nation.name"),
    columnLinage("col2", "tpch.tiny.supplier.acctbal"));
```

Please run the above "SELECT.." statement for CTAS, INSERT, UPDATE, MERGE. The last two might require more tricks to make it possible. For MERGE you should expect that statement is going to fail, but once we add support one would need to update this.